flink-staging/flink-table/src/main/scala/org/apache/flink/api/table/Table.scala

```diff
@@ -243,5 +243,29 @@ case class Table(private[flink] val operation: PlanNode) {
     this.copy(operation = Join(operation, right.operation))
   }
 
+  /**
+   * Union two[[Table]]s. Similar to an SQL UNION ALL. The fields of the two union operations
+   * must fully overlap.
+   *
+   * Example:
+   *
+   * {{{
+   *   left.unionAll(right)
+   * }}}
+   */
+  def unionAll(right: Table): Table = {
+    val leftInputFields = operation.outputFields.toSet
+    val rightInputFields = right.operation.outputFields.toSet
```

Comment:
Thanks for implementing my feedback. Unfortunately, you introduced a new bug. Can you remove the two `.toSet`? Otherwise the order of the fields will not be considered in your equality check, which is wrong. After that I will merge your PR ;-)